client-core/src/wallet.rs

```diff
@@ -10,21 +10,29 @@ use zeroize::Zeroize;
 
 use chain_core::init::address::RedeemAddress;
 
-use crate::service::{KeyService, WalletService};
-use crate::{ErrorKind, PublicKey, Result, Storage};
+use crate::service::{BalanceService, KeyService, WalletService};
+use crate::{Chain, ErrorKind, PublicKey, Result, Storage};
 
 /// Interface for a generic wallet
```

Comment:
there's not much functionality yet, and only dozes of these layered wrappers -- it's hard to know what problem this is aiming to solve. It may become clear when there's an actual implementation, but still it'd be good to document what this is supposed to be: e.g. what's the difference between `Wallet` and `WalletService` traits? 
We currently don't have a style guide on comments beyond formatting, but I'd envision something like this: https://docs.corda.net/codestyle.html#comments

> Comments that simply repeat the story already told by the code are best deleted.
> Explain what the code is doing at a higher level than is obtainable from just examining the statement and surrounding code.
> Explain why certain choices were made and the trade-offs considered.
> Explain how things can go wrong, which is a detail often not easily seen just by reading the code.

I think here the top-level comment for `Wallet` trait or the whole module would be some sort of guide for implementing this trait

Reply:
We should follow official rust documentation guidelines: https://rust-lang-nursery.github.io/api-guidelines/documentation.html

Reply:
Also, I have put a detailed explanation of each trait and struct in README.md

Reply:
Yep, the official API docs guidelines are in a similar spirit and have a convention on documenting panics etc.
I've created a task for that